### ISSUE TITLE
Per mode keybindings

### ISSUE THREAD

**Issue by tmdemelo** [jyp/boon#69]
I saw in #25 and #31 that there's some interest in having boon state keybindings in a per mode basis. I'd like to share one way of how I've implemented it. If there's enough interest in the full functionality of this, I can make a PR. Or maybe it could be put in the boon wiki.

## The idea

The following buffer local symbols are added to `emulation-mode-map-alists`: 

- One for the major-mode: `boon-major-mode-map-alist`
  - The format of its value is the same as `boon-mode-map-alist`. 
  - It updates to point to the keymap alist for the active major mode (e.g., `boon-org-mode-map-alist`) or nil.
- One for minor modes:  `boon-minor-mode-map-alist`
  - The format of its value is the same as `minor-mode-map-alist`. 
  - When there's a change of boon state, it updates to point to a corresponding keymap alist:
    - `boon-minor-mode-command-map-alist`
    - `boon-minor-mode-special-map-alist`
    - `boon-minor-mode-insert-map-alist`

The functions `boon-boonify-major-name` and `boon-boonify-minor-name` setup the extra keymaps and alists for a mode.

## Sample implementation

To keep things simple, I've done it only for command state:

[boon-boonify.el](https://gist.github.com/tmdemelo/993fb9a1615e627c8bb0bd10208be0c3#file-boon-boonify-el)

### Sample usage

With org-mode, I'd like to follow links with the Enter key when I'm in command state, but always insert newlines otherwise.

First, I run this once:

```elisp
(boon-boonify-major-name 'org-mode)
```

It will create a keymap named `boon-org-mode-command-map`. Then:

```elisp
(define-key boon-org-mode-command-map (kbd "RET") 'org-return)
(define-key org-mode-map              (kbd "RET") 'boon-newline-dwim)
```

(edit: moved code to gist; improved the idea a bit)

**Comment by vale981** [comment]
+1

**Comment by hrehfeld** [comment]
Should this be closed due to 880436e ?

**Comment by jyp** [comment]
Yes, it should. I've added a note regarding this in the documentation. Thanks!
